### ISSUE TITLE
java.lang.ArrayIndexOutOfBoundsException in `net.lingala.zip4j.headers.HeaderReader.readAesExtraDataRecord::HeaderReader.java:677` zip4j 2.9.0

### ISSUE THREAD

**Issue by ZanderHuang** [srikanth-lingala/zip4j#367]
This vulnerability is of java.lang.ArrayIndexOutOfBoundsException, and can be triggered in latest version zip4j (2.9.0).
It is caused by missing check the array index < array size and also failing to catch the runtime java exception (it should be wrapped as one kind of JSONException) and can be used for attackers to launch DoS (Denial of Service) attack for any java program that uses this library (since the user of zip4j doesn't know they need to catch this kind of exception) (CWE-129: Improper Validation of Array Index, CWE-248: Uncaught exception).
Likely, the root cause of this crash is in `net.lingala.zip4j.headers.HeaderReader.readAesExtraDataRecord::HeaderReader.java:677`.
https://github.com/srikanth-lingala/zip4j/blob/ce1cff61c019d61f5db603cba908118a3a0e3b93/src/main/java/net/lingala/zip4j/headers/HeaderReader.java#L677 The variable "aesData" has only a length of 4. When it tries to get aesData[4], it results in ArrayIndexOutOfBoundsException. 

See more detail from the following crash stack.

# Crash stack:
The crash thread's stack is as follows:

```
net.lingala.zip4j.headers.HeaderReader.readAesExtraDataRecord::HeaderReader.java:677
net.lingala.zip4j.headers.HeaderReader.readAesExtraDataRecord::HeaderReader.java:642
net.lingala.zip4j.headers.HeaderReader.readLocalFileHeader::HeaderReader.java:576
net.lingala.zip4j.io.inputstream.ZipInputStream.getNextEntry::ZipInputStream.java:91
net.lingala.zip4j.io.inputstream.ZipInputStream.getNextEntry::ZipInputStream.java:83
com.test.Entry.main::Entry.java:37
```

# Steps to reproduce: 

1. Build the following java code with the corresponding zip4j library (version 2.9.0).

```
## Download zip4j_env_reproduce.tar.gz from https://drive.google.com/file/d/1MekCBIghKxIW4j-TLjZkm8ovvLb_grm5/view?usp=sharing
tar -xf zip4j_env_reproduce.tar.gz
cd zip4j_env_reproduce
bash build.sh
```

2. Run the built program to see the crash by feeding one of the poc file contained in the pocs.tar.gz, e.g. :
(poc file can be downloaded from https://drive.google.com/file/d/1s3ixcpQ6H0O5QBNsvqBtFVrn34OhcSsX/view?usp=sharing)
```
java -jar target/Entry-1.0-SNAPSHOT-jar-with-dependencies.jar pocs/crash-05322b519e38dac6e8682c1499bbcd4f60d2fed4
```

Any further discussion for this vulnerability including fix is welcomed!

**Comment by srikanth-lingala** [comment]
Fixed with #366

**Comment by srikanth-lingala** [comment]
Fixed in v2.10.0 released today
